### ISSUE TITLE
2024.4-beta1 stuck on splash screen. 

### ISSUE THREAD

**Issue by gsture** [mullvad/mullvadvpn-app#6545]
### Is it a bug?

- [X] I know this is an issue with the app, and contacting Mullvad support is not relevant.

### I have checked if others have reported this already

- [X] I have checked the issue tracker to see if others have reported similar issues.

### Current Behavior

When opening the app there is now a splash screen but when I open the app I see the splash screen forever.  Only reopening the app brings me past the splash screen. 

### Expected Behavior

No splash screen or splash screen for a little moment. Now the app is stuck forever on splash screen. Option to disable splash screen would be welcome. 

### Steps to Reproduce

1. Open app
2. App is stuck on splash screen

### Android version

GraoheneOS 14 (AP2A.240705.005)

### Device model

Pixel 8 (tested in 2 devices)

### Mullvad VPN app version

2024.4-beta1

**Comment by Rawa** [comment]
Hello, thanks for the report! 

Does the app always get stuck in the splash screen when you open it? What is visible on the screen? When exactly does this occur?

**Comment by gsture** [comment]
> Hello, thanks for the report!
> 
> Does the app always get stuck in the splash screen when you open it? What is visible on the screen? When exactly does this occur?

I see just this:
![Screenshot_20240730-163519](https://github.com/user-attachments/assets/2aedb07b-cae2-4f87-9d2e-c7f91c7f29d6)

Then I close the app with back button/swiping back and open the app again and then it works for a while.

**Comment by gsture** [comment]
Seems like the app or deamon is crashing with memory tagging enabled. If memory tagging is disabled this problem does not occur.

See also this report where memory tagging also caused other issues:
https://github.com/mullvad/mullvadvpn-app/issues/6349

**Comment by Rawa** [comment]
Thanks for the detailed response, if you don't mind please send in a problem report after you've restarted the app after this problem occurred and we'll look into it. We have an active issue looking at MTE crashes and the logs would be of great help. 🙏

**Comment by gsture** [comment]
> Thanks for the detailed response, if you don't mind please send in a problem report mentioning this github issue after you've restarted the app after this problem occurred and we'll look into it. We have an active issue looking at MTE crashes and the logs would be of great help. 🙏

This version is way worse then all versions before with mte enabled. With previous versions the app crashed like once or twice per day but this versions stops imidiatelly after enabling mte so it should be easely reproducible on a pixel with grapheneos and mte enabled.

![Screenshot_20240731-182840~2](https://github.com/user-attachments/assets/f2c27114-26be-4855-a270-86d2d3e92e4c)

Here are 2 crash logs right after the app crashed. 1 log is from the vpn in main profile, the other crash is from the vpn in work profile. After it happens the app is stuck on splash screen but this bug is just the same as the one I mentioned in my previous comment. 

Main: 
[Log 1.txt](https://github.com/user-attachments/files/16444802/Log.1.txt)

Work:
[Log 2.txt](https://github.com/user-attachments/files/16444808/Log.2.txt)

**Comment by Rawa** [comment]
Update: 
We believe these are two separate issues. MTE crashes are still a problem but we have managed to reproduce a bug related to the new splash screen not being removed properly when the app has been killed by the system. Working on a fix for this.

**Comment by deadcombo** [comment]
Just dropping in to confirm this on the Pixel 7 Pro which does not have memory tagging.

Graphene OS 2024080200 (Android 14)

**Comment by Rawa** [comment]
Fix for this issue has been merged to main and release branch, and a new beta should be released with-in a couple of days (hopefully tomorrow), we were held up by [another issue](https://github.com/mullvad/mullvadvpn-app/pull/6567) that was uncovered while solving this.

Depending on settings of your device it is possible that this would happen (e.g certain developer options might kill you app more aggressively). Please test with the beta that will be released and create a new issue if your issue persists.

**Comment by Rawa** [comment]
Closing this issue, new release has been made [2024.4-beta2](https://github.com/mullvad/mullvadvpn-app/releases/tag/android%2F2024.4-beta2). Please re-open if you find this still relevant or create new issues if you make any more findings. 🙏
